scripts/template_upload_gce.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python2
+
+"""This script takes various parameters specified in
+cfme_data['template_upload']['template_upload_gce'] and/or by command-line arguments.
+Parameters specified by command-line have higher priority, and override data in cfme_data.
+
+This script is designed to run either as a standalone gce template uploader, or it can be used
+together with template_upload_all script. This is why all the function calls, which would
+normally be placed in main function, are located in function run(**kwargs).
+"""
+
+import argparse
+import re
+import sys
+import os
+import urllib2
+from threading import Lock, Thread
+
+from utils.conf import cfme_data
+from utils.conf import credentials
+# TODO replace by mgmt system lib
+from utils.mgmt_system import GoogleCloudSystem
+from utils.providers import list_providers
+
+lock = Lock()
```

Comment:
The lock is present in every template_upload but not used in scripts, so I assumed it is used externally.